warehouse/query-core/src/main/java/datawave/core/iterators/DatawaveFieldIndexCachingIteratorJexl.java

```diff
@@ -581,12 +584,26 @@ public void seek(Range r, Collection<ByteSequence> columnFamilies, boolean inclu
                     querySpan = ((SourceTrackingIterator) source).getQuerySpan();
                 }
 
-                // seek our underlying source to the start of the incoming range
-                // expand the range as the underlying table may not actually contain the keys in this range as we are only returning keys
-                // as specified by the returnKeyType
-                Range seekRange = new Range(lastRangeSeeked.getStartKey(), lastRangeSeeked.isStartKeyInclusive(),
-                                (lastRangeSeeked.getEndKey() == null ? null : new Key(lastRangeSeeked.getEndKey().getRow()).followingKey(PartialKey.ROW)),
-                                false);
+                // seek the source to the start of the field index that matches the specified field
+                // this initial seek ensures that data for the field is actually present
+                // if data is present, then bounding ranges are generated
+                Preconditions.checkNotNull(fiRow, "Ivarator row was null");
+                Preconditions.checkNotNull(fieldName, "Ivarator was not bounded by a field");
+
+                Range seekRange;
+                if (fieldName.toString().equals(Constants.ANY_FIELD)) {
```

Comment:
I'm not super-psyched about a carve out for testing, but might be necessary

Reply:
I'm even less psyched that we have unit tests that rely on ANYFIELD in the final query plan